sklearn/linear_model/coordinate_descent.py

```diff
@@ -92,16 +92,23 @@ def _alpha_grid(X, y, Xy=None, l1_ratio=1.0, fit_intercept=True,
 
     if Xy.ndim == 1:
         Xy = Xy[:, np.newaxis]
+
     if sparse_center:
         if fit_intercept:
             Xy -= mean_dot[:, np.newaxis]
         if normalize:
             Xy /= X_std[:, np.newaxis]
+
     alpha_max = (np.sqrt(np.sum(Xy ** 2, axis=1)).max() /
                  (n_samples * l1_ratio))
-    alphas = np.logspace(np.log10(alpha_max * eps), np.log10(alpha_max),
-                         num=n_alphas)[::-1]
-    return alphas
+
+    if alpha_max <= np.finfo(float).resolution:
```

Comment:
@agramfort out of curiosity, why do we use resolution instead of 0 directly?
( Is it for variations in the representation of floating point in different architectures so that when 0 is stored as 0.0....1, that does not affect this check? )